zebra-network/src/peer_set/initialize/tests/vectors.rs

```diff
@@ -351,16 +348,7 @@ async fn crawler_peer_limit_one_connect_ok_then_drop() {
                 connection_tracker,
             } = req;
 
-            let (server_tx, _server_rx) = mpsc::channel(0);
-            let (shutdown_tx, _shutdown_rx) = oneshot::channel();
-            let error_slot = ErrorSlot::default();
-
-            let fake_client = peer::Client {
-                shutdown_tx: Some(shutdown_tx),
-                server_tx,
-                error_slot,
-                version: Version(1),
-            };
+            let (_handle, fake_client) = MockedClientHandle::new(Version(1));
```

Comment:
Possible Coverage Reduction: How can we be sure that these changes still produce good coverage for `peer::Client` and the zebra-network async tasks?

It's probably ok, because the `PeerSet` is fake anyway, and we're just counting peers. But let's document that.